allauth/socialaccount/adapter.py

```diff
@@ -197,9 +197,22 @@ def get_app(self, request, provider):
 
         config = app_settings.PROVIDERS.get(provider, {}).get("APP")
         if config:
-            app = SocialApp(provider=provider)
-            for field in ["client_id", "secret", "key", "certificate_key"]:
-                setattr(app, field, config.get(field))
+            if (
+                app_settings.STORE_TOKENS
+                and SocialApp.objects.filter(provider=provider).exists()
+            ):
+                # We already saved the app. Fetch and return it right away...
```

Comment:
There shouldn't really be any comments here, this is the "straightforward" code path. The app exists, and we return it.